out_newrelic.go

```diff
@@ -169,43 +172,95 @@ func FLBPluginInit(ctx unsafe.Pointer) int {
 
 	config.useApiKey = len(config.apiKey) > 0
 
-	possibleeMaxBufferSize := output.FLBPluginConfigKey(ctx, "maxBufferSize")
-	if len(possibleeMaxBufferSize) == 0 {
-		config.maxBufferSize = 256000
-	} else {
-		config.maxBufferSize, _ = strconv.ParseInt(possibleeMaxBufferSize, 10, 64)
+	maxBufferSize, err := optInt64(ctx, "maxBufferSize", 256000)
+	if err != nil {
+		return output.FLB_ERROR
 	}
+	config.maxBufferSize = maxBufferSize
 
-	possibleMaxRecords := output.FLBPluginConfigKey(ctx, "maxRecords")
-	if len(possibleMaxRecords) == 0 {
-		config.maxRecords = 1024
-	} else {
-		config.maxRecords, _ = strconv.ParseInt(possibleMaxRecords, 10, 64)
+	maxRecords, err := optInt64(ctx, "maxRecords", 1024)
+	if err != nil {
+		return output.FLB_ERROR
 	}
+	config.maxRecords = maxRecords
 
-	possibleMaxTimeBetweenFlushes := output.FLBPluginConfigKey(ctx, "maxTimeBetweenFlushes")
-	if len(possibleMaxTimeBetweenFlushes) == 0 {
-		config.maxTimeBetweenFlushes = 5000
-	} else {
-		config.maxTimeBetweenFlushes, _ =  strconv.ParseInt(possibleMaxTimeBetweenFlushes, 10, 64)
+	maxTimeBetweenFlushes, err := optInt64(ctx, "maxTimeBetweenFlushes", 5000)
+	if err != nil {
+		return output.FLB_ERROR
+	}
+	config.maxTimeBetweenFlushes = maxTimeBetweenFlushes
+
+	ignoreSystemProxy, err := optBool(ctx, "ignoreSystemProxy", false)
+	if err != nil {
+		return output.FLB_ERROR
+	}
+	proxy := output.FLBPluginConfigKey(ctx, "proxy")
+
+	proxyResolver, err := getProxyResolver(ignoreSystemProxy, proxy)
+	if err != nil {
+		log.Printf("[ERROR] Invalid proxy configuration: %v", err)
+		return output.FLB_ERROR
 	}
+	config.proxyResolver = proxyResolver
 
 	bufferManager = newBufferManager(config)
 	return output.FLB_OK
 }
 
+func optInt64(ctx unsafe.Pointer, keyName string, defaultValue int64) (int64, error) {
+	rawVal := output.FLBPluginConfigKey(ctx, keyName)
+	if len(rawVal) == 0 {
+		return defaultValue, nil
+	} else {
+		value, err := strconv.ParseInt(rawVal, 10, 64)
+		if err != nil {
+			log.Printf("[ERROR] Invalid value for %s: %s. Must be an integer.", keyName, rawVal)
+			return 0, err
+		}
+		return value, nil
+	}
+}
+
+func optBool(ctx unsafe.Pointer, keyName string, defaultValue bool) (bool, error) {
+	rawVal := output.FLBPluginConfigKey(ctx, keyName)
+	if len(rawVal) == 0 {
+		return defaultValue, nil
+	} else {
+		value, err := strconv.ParseBool(rawVal)
+		if err != nil {
+			log.Printf("[ERROR] Invalid value for %s: %s. Valid values: 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.", keyName, rawVal)
```

Comment:
opinion: I think just saying "true, false" is sufficient :) No need to let Go's implementation bleed out to our config.

Reply:
OK! Fixed